taichi/backends/vulkan/aot_module_loader_impl.cpp

```diff
@@ -3,9 +3,30 @@
 #include <fstream>
 #include <type_traits>
 
+#include "taichi/backends/vulkan/runtime.h"
+
 namespace taichi {
 namespace lang {
 namespace vulkan {
+namespace {
+
+using KernelHandle = VkRuntime::KernelHandle;
+
+class KernelImpl : public AotKernel {
+ public:
+  explicit KernelImpl(VkRuntime *runtime, KernelHandle handle)
+      : runtime_(runtime), handle_(handle) {
+  }
+
+  void run(RuntimeContext *ctx) override {
+    runtime_->launch_kernel(handle_, ctx);
+  }
+
+ private:
+  VkRuntime *const runtime_;
+  const KernelHandle handle_;
```

Comment:
Out of curiosity, I wonder maybe we can ditch the KernelHandle thing and use some pure data structure to replace it?

Reply:
I do prefer `KernelHandle` for type safety. It's identical to an integer, so no runtime overhead :-)